replications/internal/queue_management.go

```diff
@@ -71,44 +81,133 @@ func (qm *durableQueueManager) InitializeQueue(replicationID platform.ID, maxQue
 		return err
 	}
 
-	// Map new durable queue to its corresponding replication stream via replication ID
-	qm.replicationQueues[replicationID] = newQueue
-
 	// Open the new queue
 	if err := newQueue.Open(); err != nil {
 		return err
 	}
 
+	// Map new durable queue and scanner to its corresponding replication stream via replication ID
+	rq := replicationQueue{
+		queue:   newQueue,
+		logger:  qm.logger,
+		done:    make(chan struct{}),
+		receive: make(chan struct{}),
+	}
+	qm.replicationQueues[replicationID] = &rq // todo does this need to be held?
+	rq.Open()
+
 	qm.logger.Debug("Created new durable queue for replication stream",
 		zap.String("id", replicationID.String()), zap.String("path", dir))
 
 	return nil
 }
 
+func (rq *replicationQueue) Open() {
+	rq.wg.Add(1)
+	go rq.run()
+}
+
+func (rq *replicationQueue) Close() error {
+	close(rq.receive)
+	close(rq.done)
+	rq.wg.Wait() // wait for goroutine to finish processing all messages
+	return rq.queue.Close()
+}
+
+func (rq *replicationQueue) run() {
+	defer rq.wg.Done()
+
+	retryInterval := time.Second // todo configurable?
+	retryTimer := time.NewTicker(retryInterval)
+	defer retryTimer.Stop()
+
+	writer := func() {
+		for {
+			_, err := rq.SendWrite(func(b []byte) error {
+				rq.logger.Info("written bytes", zap.String("bytes", string(b)))
```

Comment:
This might be a "next phase" type of question: I'm wondering about limiting the maximum number of remote writes that are in progress at a time. It seems like there should be a limit that is possibly < the amount of replication streams somebody has (maybe configurable). In that case, I think we'd want to block somewhere around here, possibly in the `SendWrite` method, if there were already the maximum amount of remote writes in progress.

If you've already thought of that / think it makes sense to wait for that until the remote HTTP writer implementation, I'm good with that too.

Reply:
Also even though this is just a placeholder, I'd probably log something like `len(b)` rather than its string representation, which will look pretty gnarly.

Reply:
I think we should probably block in this `writer func()` since blocking in `SendWrite()` wouldn't allow for the close channel to be received and leave hanging goroutines